src/bloom.cpp

```diff
@@ -34,12 +34,12 @@ CBloomFilter::CBloomFilter(unsigned int nElements, double nFPRate, unsigned int
      * Again, we ignore filter parameters which will create a bloom filter with more hash functions than the protocol limits
      * See https://en.wikipedia.org/wiki/Bloom_filter for an explanation of these formulas
      */
-    isFull(false),
     isEmpty(false),
     nHashFuncs(min((unsigned int)(vData.size() * 8 / nElements * LN2), MAX_HASH_FUNCS)),
     nTweak(nTweakIn),
     nFlags(nFlagsIn)
 {
+    isFull = vData.size() == 0;
```

Comment:
My intuition is telling me that allowing this edge condition is dangerous.  It certainly makes the correctness analysis harder.  So I went hunting for another bug and found one and a half.  First if vData.size == 0 then the Hash() function will get a DIV0 (however, the Hash() function won't be called if isFull so this is the "half" bug).

Second, if clear() is called, it will set isFull to false even if vData.size()==0, allowing the Hash() DIV0 and your out-of-bounds faults.  Yes, we could just fix clear() similarly to what you did here but...
 
I think that it makes a lot more sense to require that the minimum vData size something >= 1.  Perhaps 8 which would be a negligible # of bytes on the wire and be useful for degenerate cases where we have < 100 elements.  This ensures that we don't have DIV0s and OOB bombs hiding in the code, protected by a non-intuitive boolean, and therefore simplifies code analysis.  What are the drawbacks of this approach?

One last thing:  if UpdateEmptyFull() is called with a zero sized BloomFilter, it will be set to be simultaneously empty and full.  But in this case IDK if this is a bug... in one sense a 0 length filter IS simultaneously empty and full.  This sensible-yet-nonsensical result supports my argument that we should disallow this state.